src/SFA.DAS.ApprenticeFeedback.Application/Commands/CreateApprenticeFeedback/CreateApprenticeFeedbackHandler.cs

```diff
@@ -0,0 +1,63 @@
+﻿using MediatR;
+using SFA.DAS.ApprenticeFeedback.Domain.Interfaces;
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Linq;
+
+namespace SFA.DAS.ApprenticeFeedback.Application.Commands.CreateApprenticeFeedback
+{
+    public class CreateApprenticeFeedbackHandler : IRequestHandler<CreateApprenticeFeedbackCommand, CreateApprenticeFeedbackResponse>
+    {
+        public readonly IApprenticeFeedbackRepository _apprenticeFeedbackRepository;
+
+        public CreateApprenticeFeedbackHandler(IApprenticeFeedbackRepository apprenticeFeedbackRepository)
+        {
+            _apprenticeFeedbackRepository = apprenticeFeedbackRepository;
+        }
+
+
+        public async Task<CreateApprenticeFeedbackResponse> Handle(CreateApprenticeFeedbackCommand request, CancellationToken cancellationToken)
+        {
+
+            // Needs to retrieve apprentice feedback target based on apprentice id which should be in request
+            // May not return a single record. ( currently we assume it returns one )
+            var apprenticeFeedbackTarget = await _apprenticeFeedbackRepository.GetApprenticeFeedbackTargetById(request.ApprenticeId);
+
+            //if i can't find apprentice target record then need to throw an exception as wont be valid feedback
+            if (apprenticeFeedbackTarget == null)
+            {
+                throw new Exception("Apprentice Feedback Target not found.");
```

Comment:
We can update the exception type here, what do you think fits best?

Reply:
ArgumentNullException?